packages/env/src/__tests__/merge.ts

```diff
@@ -0,0 +1,27 @@
+import { merge } from "../merge";
+
+it("works", () => {
+  const actual = merge(
+    {
```

Comment:
Probably arrays should be added to the test if it's supposed that they should not be merged?

Reply:
I added some tests for arrays and non-plain objects.